docs/compilers/Compiler Server.md

```diff
@@ -0,0 +1,78 @@
+Compiler Server
+===============
+
+As a performance optimization, the Roslyn C# and VB compilers support using a
+server process to execute compilations instead of executing the compilation
+in-process. Engineering has focused on three performance benefits from the server:
+
+1. Re-using a single process to do compilation amortizes the cost of JITing the
+   compiler code. This is usually the single largest performance benefit.
+1. Roslyn requires assembly references to be hydrated into a rich object
+   model to perform compilation. The compiler server holds a fixed-size cache of
+   metadata symbols, which can help compilation times if most compilations share
+   a sizable portion of metadata references (e.g., mscorlib).
+1. On Windows process creation itself is expensive, so dispatching compilations
+   directly from MSBuild saves creations of csc.exe.
+
+## Usage
+
+The primary way to use the compiler server is via the
+`Microsoft.Build.Tasks.CodeAnalysis` MSBuild task, which ships in both the
+`dotnet` SDK and Visual Studio. The build task contains a build client which,
+rather than executing the `csc.exe`.
+
+If you're not using MSBuild, there's also a flag, `/shared` which can be
+passed directly to `csc.exe`, which causes `csc.exe` to behave as a client.
```

Comment:
Don't think this is clear enough. A client to what? I'd be a bit more explicit here that it's a client to the server.